### ISSUE TITLE
Text colors in editor sometimes randomly change when scrolling

### ISSUE THREAD

**Issue by bwarren18** [PowerShell/vscode-powershell#3295]
Issue Description
=====

When editing code in an editor window, on very rare occasions, the text coloring that is based on the code content type will randomly change as you scroll down the page.  To demonstrate the effect I created a bunch of comment lines between my code and as you can see even the minimap shows the problem.  Also if I copy and paste the text into another program such as an email in Outlook the random formatting is preserved.  

![image](https://user-images.githubusercontent.com/44654426/114708865-e5bb9e00-9cf9-11eb-976a-e830e991e9d3.png)
Closing and re-opening VSC fixes the issue.

Another person reported the same issue before and it was closed and suggested to be moved here: https://github.com/microsoft/vscode/issues/120800

Attached Logs
=====

Follow the instructions in the [README](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md) about
capturing and sending logs.

Environment Information
=====

Visual Studio Code
-----

| Name | Version |
| --- | --- |
| Operating System | Windows_NT x64 10.0.19042 |
| VSCode | 1.55.2|
| PowerShell Extension Version | 2021.2.2 |

PowerShell Information
-----

|Name|Value|
|---|---|
|PSVersion|5.1.19041.906|
|PSEdition|Desktop|
|PSCompatibleVersions|1.0 2.0 3.0 4.0 5.0 5.1.19041.906|
|BuildVersion|10.0.19041.906|
|CLRVersion|4.0.30319.42000|
|WSManStackVersion|3.0|
|PSRemotingProtocolVersion|2.3|
|SerializationVersion|1.1.0.1|

Visual Studio Code Extensions
-----

<details><summary>Visual Studio Code Extensions(Click to Expand)</summary>

|Extension|Author|Version|
|---|---|---|
|azurerm-vscode-tools|msazurermtools|0.15.0|
|Bookmarks|alefragnani|13.0.4|
|powershell|ms-vscode|2021.2.2|
|vscode-dotnet-runtime|ms-dotnettools|1.0.0|

**Comment by SydneyhSmith** [comment]
Thanks @bwarren18 we have seen this issue at times as well-- but it is a bit of a tricky one that will take resources to debug--thanks!

**Comment by bwarren18** [comment]
Thanks @SydneyhSmith, let me know if there is anything I can do to collect more debug information for this.

**Comment by andschwa** [comment]
@bwarren18 Does this only happen when semantic highlighting is enabled?

**Comment by bwarren18** [comment]
@andschwa Yes.  If Semantic Highlighting is set to false it looks fine; the bug reappears if set to true again.

**Comment by andschwa** [comment]
Helpful to know, thank you!

**Comment by bwarren18** [comment]
I wish I knew a way to consistently reproduce the issue.  
It seems to happen when a file is left open in VS code for a while.  
Let me know if there is any thing you would like me to check when it happens again.  Thanks.

**Comment by andschwa** [comment]
I've seen it occasionally, we have a whole project planned to address the semantic highlighting issues. It was helpful to be able to categorize this. I'm contemplating disabling semantic highlighting by default until we have it working better.

**Comment by shelby-carter** [comment]
OMG it's a horrible feature! I hate it so much! 

For others who struggle with it, just set config to: 

```json
"editor.semanticHighlighting.enabled": false,
```
